homeassistant/components/lutron_caseta/fan.py

```diff
@@ -0,0 +1,82 @@
+"""Support for Lutron Caseta fans."""
+import logging
+
+from homeassistant.components.fan import (
+    SUPPORT_SET_SPEED, FanEntity, DOMAIN)
+
+from . import LUTRON_CASETA_SMARTBRIDGE, LutronCasetaDevice
+
+_LOGGER = logging.getLogger(__name__)
+
+LUTRON_SPEED_OFF = 'Off'
```

Comment:
Only speeds in the base fan component are allowed to be returned in the speed list or accepted as set speed argument.

We can build two dicts to map between home assistant speeds and device speeds. Make a best effort. It might not be a perfect map.

Reply:
@MartinHjelmare I added the two dictionaries to map between speeds and devices (modeled off some other fan integrations). The base fan component does not have a "MediumHigh" field to map to. Not sure if you prefer that I take the approach that I did, or if you prefer that I add "mediumhigh" to the base component?

Reply:
We can't add new speeds without approval in an issue in our architecture repo.

Please remove all speeds not currently in the base fan component.

Reply:
@MartinHjelmare Understood. I will open an issue. I removed the speed not in the base fan component.